db_test.go

```diff
@@ -1776,45 +1761,46 @@ func TestVerticalCompaction(t *testing.T) {
 	}
 
 	for _, c := range cases {
-		tmpdir, err := ioutil.TempDir("", "data")
-		testutil.Ok(t, err)
+		if ok := t.Run("", func(t *testing.T) {
 
-		// Creating blocks.
-		for _, series := range c.blockSeries {
-			createBlock(t, tmpdir, 0, 0, 0, series)
-		}
+			tmpdir, err := ioutil.TempDir("", "data")
+			testutil.Ok(t, err)
 
-		// Open database.
-		db, err := Open(tmpdir, nil, nil, nil)
-		testutil.Ok(t, err)
-		testutil.Assert(t, len(db.blocks) == len(c.blockSeries), "Wrong number of blocks [before compact].")
+			for _, series := range c.blockSeries {
+				createBlock(t, tmpdir, 0, 0, 0, series)
+			}
+			db, err := Open(tmpdir, nil, nil, nil)
+			testutil.Ok(t, err)
+			db.DisableCompactions()
+			defer func() {
```

Comment:
Have a separate `defer` for the `tempdir` above (just after creating). Else if creating block or opening `db` is failed, we cannot do `os.RemoveAll(tmpdir)`.